Tests/UnitTests/Core/Detector/KdtSurfacesProviderTests.cpp

```diff
@@ -0,0 +1,104 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <boost/test/unit_test.hpp>
+
+#include "Acts/Detector/detail/KdtSurfacesProvider.hpp"
+#include "Acts/Surfaces/Surface.hpp"
+#include "Acts/Tests/CommonHelpers/CylindricalTrackingGeometry.hpp"
+#include "Acts/Utilities/Enumerate.hpp"
+
+#include <memory>
+#include <vector>
+
+using namespace Acts;
+using namespace Acts::Test;
+using namespace Acts::Experimental;
+
+GeometryContext tContext;
+CylindricalTrackingGeometry cGeometry = CylindricalTrackingGeometry(tContext);
+
+namespace {
+/// Helper mehtod that allows to use the already existing testing
+/// infrastructure with the new const-correct detector design
+///
+std::vector<std::shared_ptr<Acts::Surface>> unpackSurfaces(
+    const std::vector<const Acts::Surface*>& surfaces) {
+  std::vector<std::shared_ptr<Acts::Surface>> uSurfaces;
+  uSurfaces.reserve(surfaces.size());
+  for (const auto& s : surfaces) {
+    Surface* ncs = const_cast<Surface*>(s);
```

Comment:
will the const cast also be necessary outside the unit test?

Reply:
No, this is really only because we use the already existing Surface factory.